src/common.h

```diff
@@ -101,6 +101,19 @@
 #define PRIaddress PRIu64
 #define PRIoffset PRIzx
 
+// Footgun: len may evaluate many times, and sizes may not be evaluated at all.
+#if WABT_BIG_ENDIAN
+#define WABT_MEMCPY_ENDIAN_AWARE(dst, src, dsize, ssize, doff, soff, len)               \
```

Comment:
Does this need to be a macro, or can it be an inline function instead?

Reply:
it could be an inline function? but namespaces? this needs to be used in v128 but v128 is not in the namespace.

Reply:
(sorry, we don't really know C++. altho this does avoid evaluating `data_.size()` at all on little-endian? but compiler optimizations would probably optimize that out anyway...)